### PR TITLE
Update base ref property name

### PR DESCRIPTION
**Pull request by NiniOak**
### Description

**PROBLEM**
We found a bug where on pull request merge, the following error was returned by the `filter_changed_files_go_test.sh ` 
```
fatal: ambiguous argument 'origin/': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```
See example [here](https://github.com/hashicorp/consul/actions/runs/6172863474/job/16753960303#step:3:8) 

**SOLUTION**
Update the `conditional-skip` job with a step to conditional check the github event, 
- if event is `pull_request merged`, set GitHub branch value to `github.ref_name`
- for `pull_request` event, set GitHub branch value to `github.head_ref`
Update the `filter_changed_files_go_test.sh` script to fail on script error and set the default behaviour to trigger CI, the previous behaviour was to skip CI. 

### Testing & Reproduction steps
[CI ](https://github.com/hashicorp/consul/actions/runs/6241892960/job/16944877094) skipping tests  where all changes are from skipped directories
[CI](https://github.com/hashicorp/consul/actions/runs/6241914325/job/16944930107) running test workflows by default

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

**Review by nfi-hashicorp — APPROVED** [review]
LGTM! Assuming it works ;)

File: .github/scripts/filter_changed_files_go_test.sh
Comment (NiniOak) [reply .github/scripts/filter_changed_files_go_test.sh:32]: Good catch, I must have forgotten to change it while testing.

File: .github/scripts/filter_changed_files_go_test.sh
Comment (NiniOak) [reply .github/scripts/filter_changed_files_go_test.sh:37]: Just so I understand, you think this script should set is job set `skip-ci` which is read by dependent jobs?

Comment (zalimeni) [reply .github/scripts/filter_changed_files_go_test.sh:37]: I was thinking mainly just an inversion of the logic, nothing fundamentally different in the script - so changing the `trigger-ci == "true"` check in job defs to `skip-ci != "true"`, and in this script, defaulting the output to `false`/empty and updating it iff. all the files we check are skippable (which would still give you the clean short-circuit of the original version, I think).

**Review by zalimeni — APPROVED** [review]
Thanks so much for making these updates @NiniOak! I have a few more questions but overall I think this looks correct.

Approving to unblock you, though of course we should definitely test this out thoroughly on both a merge and PR (and any other relevant trigger) to make sure it works as expected.

Also think it's worth calling out that the less-risky version of this is getting our docs out of `consul` (cc @NicoletaPopoviciu) , which AFAIK would massively reduce the number of docs PRs - I'm guessing to the point that we might be fine w/ just running CI w/o this check, as a risk/simplicity tradeoff.

File: .github/scripts/filter_changed_files_go_test.sh
Comment (zalimeni) [reply .github/scripts/filter_changed_files_go_test.sh:8]: @NiniOak I'm also wondering whether values on the GHA event could help make this more explicit and avoid edge cases around merges... `before` or `commits` from a `push` [event](https://docs.github.com/en/webhooks/webhook-events-and-payloads#push), for example, might help when we don't have a PR?

File: .github/scripts/filter_changed_files_go_test.sh
Comment (curtbushko) [inline .github/scripts/filter_changed_files_go_test.sh:36]: **nitpick:** I am actually surprised that this works as I would have expected setting GITHUB_OUTPUT to be in a sub shell and not exported to githubs global variables.

I would personally set the environment variable setting at the workflow/job level like similar outputs.

File: .github/scripts/filter_changed_files_go_test.sh
Comment (NiniOak) [reply .github/scripts/filter_changed_files_go_test.sh:28]: No reason, was just trying to be consistent

**Review by curtbushko — APPROVED** [review]
Approving as it looks right to me.

I had one little nitpick about where to set the github variable but I won't block on it.

Note, you wanted to fully test this you could make your own fake private repo and do all of the testing there as github runners will work there...

**Comment by hc-github-team-consul-core** [comment]
Backport failed @NiniOak. Run: https://github.com/hashicorp/consul/actions/runs/6254326107

**Comment by NiniOak** [comment]
> Approving as it looks right to me.
> 
> I had one little nitpick about where to set the github variable but I won't block on it.
> 
> Note, you wanted to fully test this you could make your own fake private repo and do all of the testing there as github runners will work there...

Works on pull_request_merge see [here](https://github.com/hashicorp/consul/actions/runs/6254326231/job/16981612590)

**Pull request merged**
